R/ui.R

```diff
@@ -186,7 +186,9 @@ ui_path <- function(x, base = NULL) {
     x <- path_rel(x, base)
   }
 
-  x <- paste0(x, ifelse(is_directory, "/", ""))
+  if (!endsWith(x, "/")) {
```

Comment:
FYI `endsWith()` only appeared in R 3.3, so we can't use it w/o also using backports, which usethis does not. Right now, we still promise to support 3.2.

Anyway, I did a bit of refactoring and this is one of the things I changed.